### PR TITLE
docs: several small fixes/improvements/refactorings for the guides

### PR DESCRIPTION
**Pull request by gkalpak**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
```
[x] Documentation content changes
```

## What is the current behavior?
Some minor issues (typos, inconsistentcies, wrong/incomplete examples).

## What is the new behavior?
Less minor issues.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

Related: #21517, #21531

### CODE REVIEW

**Review by kapunahelewong — APPROVED** [review]
This is excellent. Thank you so much, @gkalpak!!

File: aio/content/guide/feature-modules.md [hunk 1]
```diff
@@ -130,5 +126,3 @@ You may also be interested in the following:
 * [Lazy Loading Modules with the Angular Router](guide/lazy-loading-ngmodules).
 * [Providers](guide/providers).
 * [Types of NgModules](guide/module-types).
```
Comment (petebacondarwin) [inline aio/content/guide/feature-modules.md]: Does the text for this link need to change to `Types of Feature Modules` ?

File: aio/content/guide/module-types.md [hunk 2]
```diff
@@ -189,4 +189,3 @@ The following table summarizes the key characteristics of each feature module gr
 You may also be interested in the following:
 * [Lazy Loading Modules with the Angular Router](guide/lazy-loading-ngmodules).
 * [Providers](guide/providers).
-* [Types of NgModules](guide/module-types).
```
Comment (petebacondarwin) [inline aio/content/guide/module-types.md]: LOL - you might also be interested in "this page"!

File: aio/content/guide/ngmodule-api.md [hunk 3]
```diff
@@ -40,192 +40,193 @@ The following table summarizes the `NgModule` metadata properties.
...
 
+      A list of [declarable](guide/ngmodule-faq#q-declarable) classes,
+      (*components*, *directives*, and *pipes*) that _belong to this module_.
 
-     A list of [declarable](guide/ngmodule-faq#q-declarable) classes,
-     (*components*, *directives*, and *pipes*) that _belong to this module_.
+      <ol>
+        <li>When compiling a template, you need to determine a set of selectors which should be used for triggering their corresponding directives.</li>
+        <li>
+          The template is compiled within a context of an `NgModule`&mdash;the `NgModule` which this template's component is declared in&mdash;which determines the set of selectors using the following rules:
```
Comment (petebacondarwin) [inline aio/content/guide/ngmodule-api.md]: "within **the** context"?

Comment (petebacondarwin) [reply aio/content/guide/ngmodule-api.md]: "the NgModule within which the template's component is declared"

File: aio/content/guide/ngmodule-api.md [hunk 4]
```diff
@@ -40,192 +40,193 @@ The following table summarizes the `NgModule` metadata properties.
...
+          <ul>
+            <li>All selectors of directives listed in `declarations`.</li>
+            <li>All exported selectors of imported `NgModules`.</li>
```
Comment (petebacondarwin) [inline aio/content/guide/ngmodule-api.md]: "All selectors of components exported from imported `NgModules`"?

File: aio/content/guide/ngmodule-api.md [hunk 5]
```diff
@@ -40,192 +40,193 @@ The following table summarizes the `NgModule` metadata properties.
...
+    <td style="vertical-align: top">
+      <code>imports</code>
+    </td>
 
- </tr>
+    <td>
 
- <tr>
+      A list of modules which should be folded into this module. Folded means it is
+      as if all of the imported NgModule properties were declared here.
```
Comment (petebacondarwin) [inline aio/content/guide/ngmodule-api.md]: "all the imported `NgModule`'s **exported** properties"

File: aio/content/guide/ngmodule-faq.md [hunk 6]
```diff
@@ -136,13 +136,13 @@ Angular doesn't like NgModules with circular references, so don't let Module 'A'
...
 
-<a id="q-reexport"></a>
+{@a q-reexport}
 
 ## What should I export?
 
 Export [declarable](guide/bootstrapping#the-declarations-array) classes that components in _other_ NgModules
 are able to reference in their templates. These are your _public_ classes.
-If you don't export a class, it stays _private_, visible only to other component
+If you don't export a declarable class, it stays _private_, visible only to other component
```
Comment (petebacondarwin) [inline aio/content/guide/ngmodule-faq.md]: other component**s**

File: aio/content/guide/frequent-ngmodules.md [hunk 7]
```diff
@@ -55,13 +55,13 @@ of some of the things they contain:
 
  <tr>
    <td><code>RouterModule</code></td>
-   <td><code>@angular/forms</code></td>
+   <td><code>@angular/router</code></td>
```
Comment (petebacondarwin) [inline aio/content/guide/frequent-ngmodules.md]: Oops :-)

File: aio/content/guide/singleton-services.md [hunk 8]
```diff
@@ -5,21 +5,19 @@
...
+An injector created from a module definition will have services which are singletons with respect to that injector. To control the lifetime of services, one controls the creation and destruction of injectors. For example, a route will have an associated module. When the route is activated, an injector is created from that module as a child of the current injector. When you navigate away from the route, the injector is destroyed. This means that services declared in a route module will have a lifetime equal to that of the route. Similarly, services provided in an application module will have the same lifetime of the application, hence singleton.
```
Comment (petebacondarwin) [inline aio/content/guide/singleton-services.md]: Surely this line should be broken into shorter lines?

**Review by petebacondarwin — APPROVED** [review]
Some nits that I think would benefit this PR but nothing to really stop it from landing.

File: aio/content/guide/singleton-services.md (hunk 8 above)
Comment (gkalpak) [reply aio/content/guide/singleton-services.md]: I wish all lines were limited to 100 chars :)

**Comment by gkalpak** [comment]
Incorporated @petebacondarwin's suggestions. CI is green. Marked for merge.

**Pull request closed without merging**